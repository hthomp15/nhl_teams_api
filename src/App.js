import React from 'react'
import './App.css';

// Components 
import Nav from './components/Nav'
import Footer from './components/Footer'

function App() {
  return (
    <div className="App">
     <Nav></Nav>
     <Footer></Footer>
    </div>
  );
}

export default App;
