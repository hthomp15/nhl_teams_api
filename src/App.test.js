import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import { rest } from "msw";
import { setupServer } from "msw/node";
import App from "./App";

const getTeamsResponse = rest.get("https://statsapi.web.nhl.com/api/v1/teams", (req, res, ctx) => {
    return res(
        ctx.json([{
            teams: [
                {
                    id: "1",
                    name: "New York Rangers",
                    venue: {
                        name: "Madison Square Garden",
                        city: "New York",
                    },
                    conference: {
                        name: "Eastern",
                    },
                    division: {
                        name: "Metropolitan",
                    },
                },
                {
                    id: "2",
                    name: "Boston Bruins",
                    venue: {
                        name: "TD Garden",
                        city: "Boston",
                    },
                    conference: {
                        name: "Eastern",
                    },
                    division: {
                        name: "Metropolitan",
                    },
                },
            ],
        }])
    );
});

const server = new setupServer(getTeamsResponse);

beforeAll(() => server.listen());
afterEach(cleanup);
afterAll(() => server.close());


test("it should load the team New York Rangers", async () => {
    render(<App />);
    const team = await screen.findByText("New York Rangers");
    expect(team).toBeInTheDocument();
});


afterEach(cleanup);

describe("App Component", () => {
    it("renders", () => {
        render(<App />);
    });
    it("matches snapshot DOM node structure", () => {
        const { asFragment } = render(<App />);
        expect(asFragment()).toMatchSnapshot();
    })
});