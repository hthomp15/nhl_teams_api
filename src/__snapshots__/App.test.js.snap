// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="nav"
    >
      <h1>
        NHL Reference
      </h1>
      <div>
        Back to Teams
      </div>
    </nav>
    <div>
      Loading...
    </div>
    <footer
      class="mt-6"
    >
      <p
        class="text-center"
      >
         Made with 
        <span
          aria-label="heart"
          role="img"
        >
          ❤️
        </span>
         by Haile Thompson, May 2022
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
